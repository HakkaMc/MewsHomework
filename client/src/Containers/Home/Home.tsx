import React from "react";
import { Typography } from "@material-ui/core";
import {
	FormattedMessage,
	injectIntl,
	WrappedComponentProps as IntlProps
} from "react-intl";
import { compose } from "recompose";

import { Search } from "../../Components";
import { messages } from "../../constants/locales/messages";

export type Props = IntlProps;

export const Home: React.FC<Props> = (props: Props) => {
	return (
		<>
			<Typography variant="h6">
				<FormattedMessage {...messages.Home.title} />
			</Typography>
			<Search
				placeholder={props.intl.formatMessage(messages.Home.placeholder)}
			/>
		</>
	);
};

export default compose<Props, any>(injectIntl)(Home);
