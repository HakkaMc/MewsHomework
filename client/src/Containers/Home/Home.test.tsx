import React from "react";
import { shallow } from "enzyme";

import { Home, Props } from "./Home";

describe("<Home/>", () => {
	let props: Props;

	beforeEach(() => {
		const intl: any = {
			formatMessage: jest.fn()
		};

		props = {
			intl
		};
	});

	it("Match snapshot", () => {
		const wrapper = shallow(<Home {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});
