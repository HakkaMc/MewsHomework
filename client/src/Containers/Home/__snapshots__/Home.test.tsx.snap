// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Match snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    variant="h6"
  >
    <FormattedMessage
      defaultMessage="Start searching a movie..."
      id="Home.title"
      values={Object {}}
    />
  </WithStyles(ForwardRef(Typography))>
  <ReduxForm
    destroyOnUnmount={false}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="search"
    getFormState={[Function]}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    submitAsSideEffect={false}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />
</Fragment>
`;
